akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -263,6 +364,232 @@ abstract class GraphStageLogic {
   final protected def isClosed[T](out: Outlet[T]): Boolean = interpreter.outStates(conn(out)) eq Closed
 
   /**
+   * Read a number of elements from the given inlet and continue with the given function,
```

Comment:
to not grow the class to large. Things that only use the bare minimum features might not need all that baggage (this might include timers, too.)